File: app/src/main/java/com/caracode/whatclothes/main/MainView.java
package com.caracode.whatclothes.main;

import net.grandcentrix.thirtyinch.TiView;
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread;
import net.grandcentrix.thirtyinch.distinctuntilchanged.DistinctUntilChanged;

public interface MainView extends TiView {

    @CallOnMainThread
    @DistinctUntilChanged
    void updateUi(final MainViewModel mainViewModel);
}

File: app/src/main/java/com/caracode/whatclothes/main/MainView.kt
package com.caracode.whatclothes.main

import io.reactivex.Observable
import net.grandcentrix.thirtyinch.TiView
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread
import net.grandcentrix.thirtyinch.distinctuntilchanged.DistinctUntilChanged

interface MainView : TiView {

    @CallOnMainThread
    @DistinctUntilChanged
    fun updateUi(mainViewModel: MainViewModel)

    fun onFabClick(): Observable<Any>

    @CallOnMainThread
    fun displayUnimplementedMessage()
}
